src/main/java/org/candlepin/resteasy/filter/VerifyAuthorizationFilter.java

```diff
@@ -251,40 +254,39 @@ protected List<Persisted> getAccessedEntities(Verify verify, Object requestValue
         }
 
         List<Persisted> entities = new ArrayList<>();
-        Class<? extends Persisted> verifyType = verify.value();
-
-        if (requestValue instanceof String) {
-            String verifyParam = (String) requestValue;
-            Persisted entity = null;
-
-            entity = storeFactory.getFor(verifyType).lookup(verifyParam);
-
-            // If the request is just for a single item, throw an exception
-            // if it is not found.
-            if (entity == null) {
-                // This is bad, we're verifying a parameter with an ID which
-                // doesn't seem to exist in the DB. Error will be thrown in
-                // invoke though.
-                String typeName = Util.getClassName(verifyType);
-                if (typeName.equals("Owner")) {
-                    typeName = i18nProvider.get().tr("Organization");
+        Class<? extends Persisted>[] verifyTypes = verify.value();
+
+        for (Class<? extends Persisted> verifyType : verifyTypes) {
+            if (requestValue instanceof String) {
+                String verifyParam = (String) requestValue;
```

Comment:
You can simplify this with a pattern variable to avoid manual casting.

```java
if (requestValue instanceof String verifyParam) {
```

Reply:
Good idea, that is much better. Updated